Assets/HoloToolkit/Input/Scripts/InputManager.cs

```diff
@@ -307,53 +301,22 @@ private void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction
             }
         }
 
-        /// <summary>
-        /// Register to gaze manager events.
-        /// </summary>
-        private void RegisterGazeManager()
-        {
-            if (!isRegisteredToGazeChanges && GazeManager.Instance != null)
-            {
-                GazeManager.Instance.FocusedObjectChanged += GazeManager_FocusedChanged;
-                isRegisteredToGazeChanges = true;
-            }
-        }
-
-        /// <summary>
-        /// Unregister from gaze manager events.
-        /// </summary>
-        private void UnregisterGazeManager()
+        private void InputSource_FocusChanged(object sender, FocusChangedEventArgs e)
         {
-            if (isRegisteredToGazeChanges && GazeManager.Instance != null)
-            {
-                GazeManager.Instance.FocusedObjectChanged -= GazeManager_FocusedChanged;
-                isRegisteredToGazeChanges = false;
-            }
-        }
+            // Create input event
+            focusEventData.Initialize(e.InputSource, e.SourceId, e.PreviousObject, e.NewObject);
 
-        private void GazeManager_FocusedChanged(GameObject previousObject, GameObject newObject)
-        {
-            if (disabledRefCount > 0)
-            {
-                return;
-            }
+            // Handler for execute events
+            ExecuteEvents.EventFunction<IFocusHandler> eventHandler = (inputHandler, eventData) => { inputHandler.OnFocusChanged(focusEventData); };
```

Comment:
See my latest changes to InputManager.cs for examples of how to avoid this anon method.

```
private static readonly ExecuteEvents.EventFunction<IFocusHandler> OnFocusChangedEventHadler =
    delegate (IFocusHandler handler, BaseEventData eventData)
    {
        InputEventData casted = ExecuteEvents.ValidateEventData<InputEventData>(eventData);
        handler.OnFocusChanged(casted);
    };
```